File: ballast-core/npm/src/application/services/impl/domain-chat-service.ts
import { inject, injectable } from "inversify";
import { TYPES as BallastCore } from "../../../dependency-injection/types";
import { IEventBus } from "../../../messaging/event-bus";
import { ChatMessageSentEvent } from "../../events";
import { IChatMessage } from "../../models/chat-message";
import { IChatService } from "../chat-service";
import { ISignInService } from "./../sign-in-service";

@injectable()
export class DomainChatService implements IChatService {

    private readonly _eventBus: IEventBus;
    private readonly _signInService: ISignInService

    public constructor(
        @inject(BallastCore.Messaging.IEventBus) eventBus: IEventBus,
        @inject(BallastCore.Application.ISignInService) signInService: ISignInService
    ) {
        this._eventBus = eventBus;
        this._signInService = signInService;
    }

    public dispose() { }

    public async sendMessageAsync(message: IChatMessage): Promise<void> {
        var player = await this._signInService.getSignedInPlayerAsync(message.fromPlayerId);
        message.fromPlayerName = player && player.name || message.fromPlayerName;
        await this._eventBus.publishAsync(ChatMessageSentEvent.fromMessage(message));
        //throw new Error("Not implemented.")
    }

}

File: ballast-core/dotnet/src/application/services/impl/DomainChatService.cs
using System;
using System.Threading.Tasks;
using Ballast.Core.Application.Events;
using Ballast.Core.Application.Models;
using Ballast.Core.Messaging;

namespace Ballast.Core.Application.Services.Impl 
{
    public class DomainChatService : IChatService
    {

        private readonly IEventBus _eventBus;
        private readonly ISignInService _signInService;

        public DomainChatService(IEventBus eventBus, ISignInService signInService)
        {
            _eventBus = eventBus;
            _signInService = signInService;
        }

        public void Dispose() { }

        public async Task SendMessageAsync(ChatMessage message) 
        {
            var player = await _signInService.GetSignedInPlayerAsync(message.FromPlayerId);
            message.FromPlayerName = player.Name;
            await _eventBus.PublishAsync(ChatMessageSentEvent.FromMessage(message));
        }
    }
}